src/ert/_c_wrappers/config/config_errors.py

```diff
@@ -0,0 +1,120 @@
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+
+class ExtJobInvalidArgsException(BaseException):
+    pass
+
+
+class ConfigWarning(UserWarning):
+    pass
+
+
+@dataclass()
+class Location:
+    filename: str
+    start_pos: Optional[int] = None
+    line: Optional[int] = None
+    column: Optional[int] = None
+    end_line: Optional[int] = None
+    end_column: Optional[int] = None
+    end_pos: Optional[int] = None
+
+
+class ConfigValidationError(ValueError):
+    def __init__(
+        self,
+        errors: str,
+        config_file: Optional[str] = None,
+        location: Optional[Union[str, Location]] = None,
+    ) -> None:
+        if config_file:
+            self.location = Location(config_file)
+        elif location is not None:
+            if isinstance(location, Location):
+                self.location = location
+            else:
+                self.location = Location(location)
+        else:
+            self.location = Location(filename="")
+
+        self.errors = errors
+
+        super().__init__(
+            (
+                f"Parsing config file `{self.config_file}` "
+                f"resulted in the errors: {self.errors}"
+            )
+            if self.config_file
+            else f"{self.errors}"
+        )
+
+    def replace(self, old_text: str, new_text: str):
+        return ConfigValidationError(
+            errors=self.errors.replace(old_text, new_text),
+            config_file=self.config_file,
+            location=self.location,
+        )
+
+    @property
+    def config_file(self):
+        return self.location.filename
+
+    @config_file.setter
+    def config_file(self, config_file):
+        self.location.filename = config_file
+
+    def get_error_messages(self):
+        return [self.errors]
+
+
+class CombinedConfigError(ConfigValidationError):
+    def __init__(
+        self,
+        errors: Optional[
+            List[Union[ConfigValidationError, "CombinedConfigError"]]
+        ] = None,
+    ):
+        self.errors = []
```

Comment:
This assigns to the superclass' field errors, which has a different type annotation

Reply:
An idea: Could you, instead of making this class a sort of container, simply make a list of the errors, add to that list, and in the end construct a CombinedConfigError? Then you can just assign the correct string to this field as well.

Reply:
> This assigns to the superclass' field errors, which has a different type annotation

Oops! Changing name to `errors_list` to clear up the variable overshadowing